app/models/agents/rss_agent.rb

```diff
@@ -105,6 +106,10 @@ def validate_options
         errors.add(:base, "Please provide 'expected_update_period_in_days' to indicate how many days can pass without an update before this Agent is considered to not be working")
       end
 
+      if options['max_ids'].present? && options['max_ids'].to_i < 1
```

Comment:
> Is it maybe the case that the options hash can have values with liquid formatting and the interpolated hash has the real values after the liquid formatting has been interpreted?

Yes that is exactly what happens. You are right that we are not very consistent with the validations and usage of interpolated (most Agents use interpolated in the code but not in the validations). When `interpolated` is used in `check`, validating `options` should only check for the presence.
The example you gave `"seen_ids_to_keep" : "{{ urls | size | times : 15}}"` will not pass the validation because `to_i` will convert the string to `0`. So you could either remove the integer validation or use `interpolated`, I don't have a strong opinion about it.